brayns/Brayns.cpp

```diff
@@ -91,35 +90,28 @@ struct Brayns::Impl
         BRAYNS_INFO << std::endl;
 
         BRAYNS_INFO << "Parsing command line options" << std::endl;
-        _parametersManager->parse(argc, argv);
-        _parametersManager->print();
+        _parametersManager.parse(argc, argv);
+        _parametersManager.print();
 
-        // Initialize keyboard handler
-        _keyboardHandler.reset(new KeyboardHandler());
         _registerKeyboardShortcuts();
 
-        // Get rendering engine
-        _engineFactory.reset(
-            new EngineFactory(argc, argv, *_parametersManager));
         createEngine();
 
 #if (BRAYNS_USE_DEFLECT || BRAYNS_USE_NETWORKING)
+        // after createEngine() to execue in parallel to scene loading
         _extensionPluginFactory.reset(
-            new ExtensionPluginFactory(*_parametersManager));
+            new ExtensionPluginFactory(_parametersManager));
 #endif
+
+        if (!isAsyncMode())
+            _dataLoadingFuture.get();
     }
 
     void createEngine()
     {
-        if (_engine)
-        {
-            _engineFactory->remove(_engine);
-            _engine.reset();
-        }
-
         const auto& engineName =
-            _parametersManager->getRenderingParameters().getEngine();
-        _engine = _engineFactory->get(engineName);
+            _parametersManager.getRenderingParameters().getEngine();
+        _engine = _engineFactory.create(engineName);
```

Comment:
I would still _engine.reset() explicitly, otherwise two engines are active until the assignment to the unique_ptr is done